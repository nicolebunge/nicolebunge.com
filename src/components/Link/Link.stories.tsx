import Link from '.';

export default {
  component: Link,
};

export const Default = {
  args: {
    children: 'Internal Link',
    to: '/',
  },
};

export const External = {
  args: {
    children: 'External Link',
    to: 'http://example.com',
  },
};
