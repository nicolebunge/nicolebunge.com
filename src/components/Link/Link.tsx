/* eslint-disable jsx-a11y/anchor-has-content */

import { Link as GatsbyLink } from 'gatsby';
import React, { ReactNode } from 'react';

export interface LinkProps extends React.AnchorHTMLAttributes<HTMLAnchorElement> {
  activeClassName?: string;
  children: ReactNode;
  external?: boolean;
  to: string;
}

function Link(props: LinkProps): JSX.Element {
  const { to, external = false, ...otherProps } = props;

  if (external || to.startsWith('http')) {
    return <a href={to} target="_blank" rel="noopener noreferrer" {...otherProps} />;
  }

  return <GatsbyLink to={to} {...otherProps} />;
}

export default Link;
