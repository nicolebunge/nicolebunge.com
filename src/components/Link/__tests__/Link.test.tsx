import { composeStories } from '@storybook/testing-react';
import { render } from '@testing-library/react';
import React from 'react';
import * as stories from '../Link.stories';

const { Default, External } = composeStories(stories);

describe('Link component', () => {
  it('renders correctly', () => {
    const { container } = render(<Default />);

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render external link correctly', () => {
    const { container } = render(<External />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
