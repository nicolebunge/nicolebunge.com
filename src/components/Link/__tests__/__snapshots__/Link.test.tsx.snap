// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link component renders correctly 1`] = `
<a
  href="/"
>
  Internal Link
</a>
`;

exports[`Link component should render external link correctly 1`] = `
<a
  href="http://example.com"
  rel="noopener noreferrer"
  target="_blank"
>
  External Link
</a>
`;
